src/binary-reader.cc

```diff
@@ -466,6 +466,17 @@ static void write_indent(LoggingContext* ctx) {
     LOGF_NOINDENT(__VA_ARGS__); \
   } while (0)
 
+static void logging_on_error(BinaryReaderContext* context,
```

Comment:
Hmm.. i just removed this didn't I?  Why does it crash? 

Perhaps we need more tests with logging enabled.   Can you make this a separate PR maybe?